src/OpenSearch/HttpTransport.php

```diff
@@ -38,7 +40,14 @@ public function sendRequest(
     ): array|string|null {
         $request = $this->createRequest($method, $uri, $params, $body, $headers);
         $response = $this->client->sendRequest($request);
-        return $this->serializer->deserialize($response->getBody()->getContents(), $response->getHeaders());
+        $statusCode = $response->getStatusCode();
+        $responseBody = $response->getBody()->getContents();
+        $responseHeaders = $response->getHeaders();
+        if ($statusCode >= 400) {
```

Comment:
For later, is the client handling 30x, like redirects?

Reply:
Both Guzzle and Symfony HTTP Client follow redirects.